mirrors_update.py

```diff
@@ -0,0 +1,318 @@
+    :param repos: the list of repos which should be provided by mirrors
+    :param mirrorlist_dir: the directory which contains mirrorlist files
+                           per an each version
+    """
+
+    for mirror_info in verified_mirrors:
+        if mirror_info['status'] != 'ok':
+            logging.warning(
+                'Mirror "%s" is expired and isn\'t added to mirrorlist',
+                mirror_info['name']
+            )
+            continue
+        addresses = mirror_info['address']
+        for version in versions:
+            version_dir = os.path.join(
+                mirrorlist_dir,
+                str(version),
+            )
+            os.makedirs(version_dir, exist_ok=True)
+            for repo_info in repos:
+                mirror_url = next(iter([
+                    address for protocol_type, address in addresses.items()
+                    if protocol_type in REQUIRED_MIRROR_PROTOCOLS
+                ]))
+                full_mirror_path = os.path.join(
+                    mirror_url,
+                    str(version),
+                    repo_info['path'],
+                )
+                mirrorlist_path = os.path.join(
+                    version_dir,
+                    repo_info['name'],
+                )
+                with open(mirrorlist_path, 'a') as mirrorlist_file:
+                    mirrorlist_file.write(f'{full_mirror_path}\n')
+
+
+def generate_mirrors_table(
+    mirrors_table_path: AnyStr,
+    verified_mirrors: List[Dict[AnyStr, Union[Dict, AnyStr]]],
+) -> None:
+    """
+    Generates mirrors table from list verified mirrors
+    :param mirrors_table_path: path to file with mirrors table
+    :param verified_mirrors: list of verified mirrors
+    """
+    columns_names = (
+        'Name',
+        'Sponsor',
+        'Status',
+        *(
+            protocol.upper() for protocol in ALL_MIRROR_PROTOCOLS
+        ),
+    )
+
+    header_separator = f"| {' | '.join(['---'] * len(columns_names))} |"
+    table_header = f"| {' | '.join(columns_names)} |\n{header_separator}"
+    address_prefixes = defaultdict(lambda: 'Link')
+    address_prefixes.update({
+        'https': 'Mirror',
+        'http': 'Mirror',
+        'rsync': 'Link',
+    })
+    with open(mirrors_table_path, 'a') as mirrors_table_file:
+        logging.info('Generate mirrors table')
+        mirrors_table_file.write(f'{table_header}\n')
+        for mirror_info in verified_mirrors:
+            logging.info(
+                'Adding mirror "%s" to mirrors table',
+                mirror_info['name']
+            )
+            addresses = mirror_info['address']
+            for protocol in ALL_MIRROR_PROTOCOLS:
+                if protocol in addresses:
+                    link = f'[{address_prefixes[protocol]}]' \
+                           f'({addresses[protocol]})'
+                else:
+                    link = ''
+                mirror_info[f'{protocol}_link'] = link
```

Comment:
if link ends with '/' symbol we should remove it due to vuepress behavior when processing rsync:// links
but it'll never hurts to remove '/' from http/https addresses so it can be done for every link